packages/flutter/test/material/time_picker_test.dart

```diff
@@ -358,989 +234,1240 @@ void _tests() {
-    await mediaQueryBoilerplate(tester, false);
-
-    final CustomPaint dialPaint = tester.widget(findDialPaint);
-    final dynamic dialPainter = dialPaint.painter;
-    // ignore: avoid_dynamic_calls
-    final List<dynamic> primaryLabels = dialPainter.primaryLabels as List<dynamic>;
-    // ignore: avoid_dynamic_calls
-    expect(primaryLabels.map<String>((dynamic tp) => tp.painter.text.text as String), labels12To11);
-
-    // ignore: avoid_dynamic_calls
-    final List<dynamic> secondaryLabels = dialPainter.secondaryLabels as List<dynamic>;
-    // ignore: avoid_dynamic_calls
-    expect(secondaryLabels.map<String>((dynamic tp) => tp.painter.text.text as String), labels12To11);
-  });
-
-  testWidgets('respects MediaQueryData.alwaysUse24HourFormat == true', (WidgetTester tester) async {
-    await mediaQueryBoilerplate(tester, true);
-
-    final CustomPaint dialPaint = tester.widget(findDialPaint);
-    final dynamic dialPainter = dialPaint.painter;
-    // ignore: avoid_dynamic_calls
-    final List<dynamic> primaryLabels = dialPainter.primaryLabels as List<dynamic>;
-    // ignore: avoid_dynamic_calls
-    expect(primaryLabels.map<String>((dynamic tp) => tp.painter.text.text as String), labels00To22);
-
-    // ignore: avoid_dynamic_calls
-    final List<dynamic> secondaryLabels = dialPainter.secondaryLabels as List<dynamic>;
-    // ignore: avoid_dynamic_calls
-    expect(secondaryLabels.map<String>((dynamic tp) => tp.painter.text.text as String), labels00To22);
-  });
-
-  testWidgets('provides semantics information for AM/PM indicator', (WidgetTester tester) async {
-    final SemanticsTester semantics = SemanticsTester(tester);
-    await mediaQueryBoilerplate(tester, false);
-
-    expect(
-      semantics,
-      includesNodeWith(
-        label: 'AM',
-        actions: <SemanticsAction>[SemanticsAction.tap],
-        flags: <SemanticsFlag>[
-          SemanticsFlag.isButton,
-          SemanticsFlag.isChecked,
-          SemanticsFlag.isInMutuallyExclusiveGroup,
-          SemanticsFlag.hasCheckedState,
-          SemanticsFlag.isFocusable,
-        ],
-      ),
-    );
-    expect(
-      semantics,
-      includesNodeWith(
-        label: 'PM',
-        actions: <SemanticsAction>[SemanticsAction.tap],
-        flags: <SemanticsFlag>[
-          SemanticsFlag.isButton,
-          SemanticsFlag.isInMutuallyExclusiveGroup,
-          SemanticsFlag.hasCheckedState,
-          SemanticsFlag.isFocusable,
-        ],
-      ),
-    );
+  group('Time picker - Dialog', () {
+    testWidgets('Material3 has Sentence case labels', (WidgetTester tester) async {
+      await startPicker(tester, (TimeOfDay? time) {
+        expect(find.text('Select time'), findsOneWidget);
+        expect(find.text('Enter time'), findsOneWidget);
+        expect(find.text('Cancel'), findsOneWidget);
+      }, useMaterial3: true);
+    });
 
-    semantics.dispose();
-  });
+    testWidgets('Material2 has ALL CAPS labels', (WidgetTester tester) async {
```

Comment:
For M2 specific tests it would be good to hardwire `useMaterial3: false` so that when we change the default to true they will still work properly.

Reply:
Oh, good point. Done.